packages/expo-notifications/src/Mailbox.ts

```diff
@@ -0,0 +1,77 @@
+import { EventEmitter as NativeEventEmitter, NativeModulesProxy } from '@unimodules/core';
+import {
+  ForegroundNotification,
+  OnUserInteractionListener,
+  OnForegroundNotificationListener,
+  UserInteraction,
+  OnTokenChangeListener,
+  TokenMessage,
+} from './Notifications.types';
+
+const { ExpoNotifications } = NativeModulesProxy;
+const DeviceEventEmitter = new NativeEventEmitter(ExpoNotifications);
+
+export class Mailbox {
+  private onUserInteractionListeners: Map<string, OnUserInteractionListener>;
+  private onForegroundNotificationListeners: Map<string, OnForegroundNotificationListener>;
+  private onTokenChangeListener: OnTokenChangeListener | null;
+
+  constructor() {
+    this.onUserInteractionListeners = new Map();
+    this.onForegroundNotificationListeners = new Map();
+    this.onTokenChangeListener = null;
+    DeviceEventEmitter.addListener(
+      'Expo.onUserInteraction',
+      this.onUserInteraction.bind(this)
+    );
+    DeviceEventEmitter.addListener(
+      'Expo.onForegroundNotification',
+      this.onForegroundNotification.bind(this)
+    );
+    DeviceEventEmitter.addListener(
+      'Expo.onTokenChange',
+      this.onTokenChange.bind(this)
+    );
+  }
+
+  addOnUserInteractionListener(listenerName: string, listener: OnUserInteractionListener) {
+    this.onUserInteractionListeners.set(listenerName, listener);
```

Comment:
Soo, when you'll be changing this logic for registering `listeners` for this method (same applied to others) then, please use for example some `internal-auto-incrementing-IDs-generator`  that would allow sth like this:
```ts
// globalScope of this file or class, because listeners are registered per class
onUserInteractionListenersCounter = 0;

addOnUserInteractionListener(listener: OnUserInteractionListener): { remove: () => void } { 
  const listenerID = this.onUserInteractionListenersCounter++;
  this.onUserInteractionListeners.set(listenerID, listener);
  return {
    remove: () => { this.onUserInteractionListeners.delete(listenerID) },
  };
}
```
Maybe we can even go with signature
```ts
addOnUserInteractionListener(listener: OnUserInteractionListener): () => void { 
```
but this way we wouldn't tell what that function actually does, so maybe let's stick with first suggestion.